functions/_fzf_search_completions.fish

```diff
@@ -0,0 +1,189 @@
+# External limitations:
+# - fish-shell/fish-shell#9577: folders ending in '=' break fish's internal path completion
+#   (It thinks that it has to complete from / because `a_folder=/` looks like an argument to fish.
+# - fish cannot give reliable context information on completions. Knowing whether a completion
+#   is a file or argument can only be determined via a heuristic.
+# - Completion is slow. Not because this script is slow but because fish's `complete` command
+#   can take several seconds even for just 100 entries. Just run `time complete --escape --do-complete l`
+#   to see for yourself.
+set --global _fzf_search_completions_min_description_offset 14
+function _fzf_search_completions --description "Shell completion using fzf"
+	# Produce a list of completions from which to choose
+	# and do nothing if there is nothing to select from
+	set -l cmd (commandline -co) (commandline -t)
+	set -l completions (complete --escape --do-complete "$cmd")
+	test (count $completions) -eq 0; and return
```

Comment:
this would read better (and be slightly more performant) as `test -z $completions; and return`. Might need `--`, not sure

Reply:
I have to disagree here, since `test -z $completions` actually has different semantics and that would obscure the original intention (which actually is to test whether the array has any members).

You would also need to add quotes like `test -z "$completions"` to prevent passing the array entries as separate parameters `test -z entry1 entry2` is an invalid command, but then there is this edge case: `set arr ""; test -z "$arr"` is true even though the array is not actually empty.